data/sdss5.cfg

```diff
@@ -240,21 +240,21 @@ conflist = $BOSS_SPECTRO_REDUX/{run2d}/conflist.fits
 fieldlist = $BOSS_SPECTRO_REDUX/{run2d}/fieldlist-{run2d}.fits
 spAll = $BOSS_SPECTRO_REDUX/{run2d}/spAll-{run2d}.fits
 spAllLine = $BOSS_SPECTRO_REDUX/{run2d}/spAllLine-{run2d}.fits
-spArc = $BOSS_SPECTRO_REDUX/{run2d}/{fieldid}{isplate}/spArc-{br}{id}-{frame:0>8}.fits.gz
-spCFrame = $BOSS_SPECTRO_REDUX/{run2d}/{fieldid}{isplate}/spFrame-{br}{id}-{frame:0>8}.fits.gz
-spFlat = $BOSS_SPECTRO_REDUX/{run2d}/{fieldid}{isplate}/spFlat-{br}{id}-{frame:0>8}.fits.gz
-spFluxcalib = $BOSS_SPECTRO_REDUX/{run2d}/{fieldid}{isplate}/spFluxcalib-{br}{id}-{frame:0>8}.fits.gz
-spFluxcorr = $BOSS_SPECTRO_REDUX/{run2d}/{fieldid}{isplate}/spFrame-{br}{id}-{frame:0>8}.fits.gz
-spFluxdistort = $BOSS_SPECTRO_REDUX/{run2d}/{fieldid}{isplate}/spFluxdistort-{fieldid}-{mjd}.fits
-spFrame = $BOSS_SPECTRO_REDUX/{run2d}/{fieldid}{isplate}/spFrame-{br}{id}-{frame:0>8}.fits.gz
-spPlate = $BOSS_SPECTRO_REDUX/{run2d}/{fieldid}{isplate}/spPlate-{fieldid}-{mjd}.fits
-spZall = $BOSS_SPECTRO_REDUX/{run2d}/{fieldid}{isplate}/{run1d}/spZall-{fieldid}-{mjd}.fits
-spZbest = $BOSS_SPECTRO_REDUX/{run2d}/{fieldid}{isplate}/{run1d}/spZbest-{fieldid}-{mjd}.fits
-spZline = $BOSS_SPECTRO_REDUX/{run2d}/{fieldid}{isplate}/{run1d}/spZline-{fieldid}-{mjd}.fits
-spXCSAO = $BOSS_SPECTRO_REDUX/{run2d}/{fieldid}{isplate}/{run1d}/spXCSAO-{fieldid}-{mjd}.fits
-spSpec = $BOSS_SPECTRO_REDUX/{run2d}/{fieldid}{isplate}/coadd/{mjd}/spSpec-{fieldid}-{mjd}-{catalogid}.fits
-specLite = $BOSS_SPECTRO_REDUX/{run2d}/spectra/lite/{fieldid}{isplate}/{mjd}/spec-{fieldid}-{mjd}-{catalogid}.fits
-specFull = $BOSS_SPECTRO_REDUX/{run2d}/spectra/full/{fieldid}{isplate}/{mjd}/spec-{fieldid}-{mjd}-{catalogid}.fits
+spArc = $BOSS_SPECTRO_REDUX/{run2d}/@pad_fieldid|{fieldid}@isplate|/spArc-{br}{id}-{frame:0>8}.fits.gz
+spCFrame = $BOSS_SPECTRO_REDUX/{run2d}/@pad_fieldid|{fieldid}@isplate|/spFrame-{br}{id}-{frame:0>8}.fits.gz
```

Comment:
`@pad_fieldid|{fieldid}` looks a little unwieldy to me.  Rather than define a function that sits it front of `fieldid`, you could probably have the function return either the `fieldid` or padded `fieldid`, depending on the conditions?  Then you could replace instances of `@pad_fieldid|{fieldid}` with `@pad_fieldid|` and simplify the path a bit.